src/Calculator/Views/DateCalculator.xaml.cpp

```diff
@@ -186,7 +186,21 @@ void DateCalculator::SetDefaultFocus()
 void DateCalculator::DateCalcOption_Changed(_In_ Platform::Object ^ sender, _In_ Windows::UI::Xaml::Controls::SelectionChangedEventArgs ^ e)
 {
     FindName("AddSubtractDateGrid");
-    DateCalculationOption->SelectionChanged -= m_dateCalcOptionChangedEventToken;
+    auto dateCalcViewModel = safe_cast<DateCalculatorViewModel ^>(this->DataContext);
+
+    // https://github.com/microsoft/calculator/issues/254
+    // From Date Field needs to persist across Date Difference and Add Substract Date Mode.
+    // So when the mode dropdown changes, update the other datepicker with the latest date.
+    if (dateCalcViewModel->IsDateDiffMode)
+    {
+        IBox<DateTime> ^ addSubtractModeFromDate = ref new Box<DateTime>(AddSubtract_FromDate->Date->Value);
```

Comment:
`AddSubtract_FromDate->Date` is already an `IBox<DateTime> ^`, you can simplify the code:
`DateDiff_FromDate->Date = AddSubtract_FromDate->Date`

Reply:
I was able to refactor a bit down to DateDiff_FromDate->Date = AddSubtract_FromDate->Date->Value instead of having the extra line. But testing my changes made me aware of a null pointer exception. In the case of never having set a date and switching to the other date, Date would never have been set, and accessing the value was throwing a null pointer exception. 

Fixed this by only setting the value if the date isn't null. Should see the updated changes when I recommit. I also saw some tests failing, maybe this was the culprit.